src/data_provider/src/packages/pkgWrapper.h

```diff
@@ -125,25 +132,50 @@ class PKGWrapper final : public IPackageWrapper
                 }
             };
 
+            static const auto truncateVersion
+            {
+                [](const std::string & val)
+                {
+                    constexpr auto VERSION_FORMAT{R"(^([^.]+\.[^.]+\.[^.]+).*$)"};
+                    static std::regex versionFormatRegex{VERSION_FORMAT};
+                    std::string version;
+                    Utils::findRegexInString(val, version, versionFormatRegex, 1);
+                    return version;
+                }
+            };
+
             const auto getDataFnc
             {
                 [this, &filePath](std::istream & data)
                 {
                     std::string line;
 
+                    m_name = UNKNOWN_VALUE;
+                    m_version = UNKNOWN_VALUE;
+                    m_installTime = UNKNOWN_VALUE;
+                    m_location = filePath;
+                    m_groups = UNKNOWN_VALUE;
+                    m_description = UNKNOWN_VALUE;
+                    m_size = 0;
+                    m_priority = UNKNOWN_VALUE;
+                    m_multiarch = UNKNOWN_VALUE;
+                    m_source = filePath.find(UTILITIES_FOLDER) ? "utilities" : "applications";
+
                     while (std::getline(data, line))
                     {
                         line = Utils::trim(line, " \t");
 
-                        if (line == "<key>CFBundleName</key>" &&
+                        if ((line == "<key>CFBundleName</key>" ||
+                                line == "<key>PackageIdentifier</key>") &&
                                 std::getline(data, line))
                         {
                             m_name = getValueFnc(line);
                         }
-                        else if (line == "<key>CFBundleShortVersionString</key>" &&
+                        else if ((line == "<key>CFBundleShortVersionString</key>" ||
+                                  line == "<key>PackageVersion</key>") &&
                                  std::getline(data, line))
                         {
-                            m_version = getValueFnc(line);
+                            m_version = truncateVersion(getValueFnc(line));
```

Comment:
@MiguelazoDS Please do not truncate the information obtained from the vendor, even if it is extensive.

Reply:
Removed